compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java

```diff
@@ -154,12 +221,97 @@ private ObjectConstructorExpressionNode modifyServiceObjectNode(ObjectConstructo
+        modifier = modifier.withAnnotations(annotations);
+        NodeList<Node> members = node.members();
+        if (this.isSubgraph) {
+            members = addServiceResourceResolverToMembers(members, node.location());
+            if (this.entities.size() > 0) {
+                members = addEntityResourceResolverToMembers(members, node.location());
+            }
+        }
+        modifier = modifier.withMembers(members);
+        return modifier.apply();
     }
 
     private ServiceDeclarationNode modifyServiceDeclarationNode(ServiceDeclarationNode node, String schemaString) {
         MetadataNode metadataNode = getMetadataNode(node, schemaString);
-        return node.modify().withMetadata(metadataNode).apply();
+        ServiceDeclarationNode.ServiceDeclarationNodeModifier modifier = node.modify();
+        modifier = modifier.withMetadata(metadataNode);
+        NodeList<Node> members = node.members();
+        if (this.isSubgraph) {
+            members = addServiceResourceResolverToMembers(members, node.location());
+            if (this.entities.size() > 0) {
+                members = addEntityResourceResolverToMembers(members, node.location());
+            }
+        }
+        modifier = modifier.withMembers(members);
+        return modifier.apply();
+    }
+
+    private NodeList<Node> addEntityResourceResolverToMembers(NodeList<Node> serviceMembers, Location location) {
+        FunctionDefinitionNode entityResolver = getEntityResolver(location);
+        return serviceMembers.add(entityResolver);
+    }
+
+    private NodeList<Node> addServiceResourceResolverToMembers(NodeList<Node> serviceMembers, Location location) {
+        FunctionDefinitionNode serviceResolver = getServiceResolver(location);
+        return serviceMembers.add(serviceResolver);
+    }
+
+    private FunctionDefinitionNode getEntityResolver(Location location) {
+        try {
+            String mapDef = getEntityTypedescMapInitializer();
+            Class<GraphqlSourceModifier> currentClass = GraphqlSourceModifier.class;
+            InputStream inputStream = currentClass.getResourceAsStream("/entity_resolver.bal.partial");
+            String entityResolver = readFromInputStream(inputStream);
+            Objects.requireNonNull(inputStream).close();
+            entityResolver = entityResolver.replaceAll(ENTITY, this.entityUnionTypeName);
+            entityResolver = entityResolver.replaceAll(MAP_DEFINITION_PLACEHOLDER, mapDef);
+            if (this.subgraphModulePrefix != null) {
+                entityResolver = entityResolver.replaceAll(SUBGRAPH_MODULE_PLACEHOLDER, this.subgraphModulePrefix);
+            }
+            return (FunctionDefinitionNode) NodeParser.parseObjectMember(entityResolver);
+        } catch (IOException | NullPointerException e) {
+            updateContext(this.context, location, CompilationDiagnostic.FAILED_TO_ADD_ENTITY_RESOLVER, e.getMessage());
+            return null;
+        }
```

Comment:
I still do not understand the reason for not using Ballerina SyntaxAPI to generate the functions, which for me seems like the safer option. Kinda like coding in notepad vs. coding in an IDE. WDYT?

Reply:
As we discussed in the offline call, we agreed to keep this implementation because the resolver logic in the file is easy to read, and we do not need to worry about code formatting as this generated code is not visible to the user.